gslib/tests/test_hmac.py

```diff
@@ -357,3 +403,168 @@ def test_hmac_fails_for_xml(self):
         stderr = self.RunGsUtil(command, expected_status=1, return_stderr=True)
         self.assertIn(
             'The "hmac" command can only be used with the GCS JSON API', stderr)
+
+
+class TestHmacUnit(testcase.GsUtilUnitTestCase):
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
```

Comment:
We normally just patch `RunCommand`
For example, in `test_hash.py`
```
mock.patch.object(hash.HashCommand, 'RunCommand', new=mock.Mock())
```

Reply:
Changed all unit tests for the shim translation using this suggestion, removed code that mocked the API.